src/Rules.Framework/Builder/RuleBuilderExtensions.cs

```diff
@@ -25,39 +25,30 @@ public static class RuleBuilderExtensions
             Operators condOperator,
             TDataType operand)
         {
-            var valueCondition = new ConditionNodeBuilder<TConditionType>()
-                .AsValued(conditionType)
-                .OfDataType<TDataType>()
-                .WithComparisonOperator(condOperator)
-                .SetOperand(operand)
-                .Build();
+            var rootConditionNodeBuilder = new RootConditionNodeBuilder<TConditionType>();
+
+            var valueCondition = rootConditionNodeBuilder.Value(conditionType, condOperator, operand);
 
             return ruleBuilder.WithCondition(valueCondition);
         }
 
         /// <summary>
-        /// Sets the new rule with the specified composed condition.
+        /// Sets the new rule with the specified root condition.
         /// </summary>
         /// <typeparam name="TContentType">The type of the content type.</typeparam>
         /// <typeparam name="TConditionType">The type of the condition type.</typeparam>
         /// <param name="ruleBuilder">The rule builder.</param>
-        /// <param name="logicOperator">The operator.</param>
         /// <param name="conditionFunc">The condition func.</param>
         /// <returns></returns>
-        public static IRuleBuilder<TContentType, TConditionType> WithConditions<TContentType, TConditionType>(
+        public static IRuleBuilder<TContentType, TConditionType> WithCondition<TContentType, TConditionType>(
             this IRuleBuilder<TContentType, TConditionType> ruleBuilder,
-            LogicalOperators logicOperator,
-            Func<IComposedConditionNodeBuilder<TConditionType>, IComposedConditionNodeBuilder<TConditionType>> conditionFunc)
+            Func<IRootConditionNodeBuilder<TConditionType>, IConditionNode<TConditionType>> conditionFunc)
```

Comment:
I agree with using an extension method, but I think we should change the namespace at the beginning of this file to `Rules.Framework` only. Otherwise, the library consumers will have to include the namespace `Rules.Framework.Builder` to build using this new approach.

Reply:
And maybe this method could afterwards be integrated as part of the RulesBuilder itself (later once we remove the obsolete parts). But even the RulesBuilder currenly is in the Rules.Framework.Builder namespace.

Reply:
Yes, I agree with adding to the `RuleBuiler` itself. As a matter of fact, I think you should consider doing that already 😉

About the namespace, when I suggested the change, it was exactly for this extension method and not the whole folder and its files. Just consider why it is necessary: the base "RuleBuiler" when you are starting to create a rule with fluent API is on base namespace `Rules.Framework`, but the subsequent methods on that fluent API are in fact on the namespace `Rules.Framework.Builer`. Because those methods are included already on the interfaces used by the Rule builder fluent API, you don't need (as a consumer) to include the `Rules.Framework.Builer` namespace to be able to use them. But extension methods are not the case, because they are not naturally presented to the consumer if, for example, using Visual Studio with IntelliSense - the consumer needs to include the namespace to be able to use the extension method, and that's not a very good experience to the consumer in my opinion (as the consumer needs to know that he needs to include the namespace). Maybe I'm nitpicking again with details, but I think a lot about how consumers will use those APIs.

So, TL;DR version: you can either change the namespace of this extensions or move the extensions to the actual `RuleBuilder` - any of these approaches resolves the issue I was seeing here 🙂
Let me know what you think!

Reply:
But (as I was trying to say before), if I understand correctly, both the IRuleBuilder and the RuleBuilder are on the namespace Rules.Framework.Builder, as you can see here 🙃 
* https://github.com/Farfetch/rules-framework/blob/master/src/Rules.Framework/Builder/IRuleBuilder.cs
* https://github.com/Farfetch/rules-framework/blob/master/src/Rules.Framework/Builder/RuleBuilder.cs

Reply:
Ok, I've found there is another RuleBuilder class that I didn't know about 😆 
* https://github.com/Farfetch/rules-framework/blob/master/src/Rules.Framework/RuleBuilder.cs 

I'll think about that tomorrow and then let you know! 👍🏼

Reply:
This one is now resolved the way things were refactored ✅